cloud/amazon/ec2_vpc_nat_gateway.py

```diff
@@ -0,0 +1,1070 @@
+            }
+        ]
+
+    try:
+        if not check_mode:
+            gateways = client.describe_nat_gateways(**params)['NatGateways']
+            if gateways:
+                for gw in gateways:
+                    existing_gateways.append(convert_to_lower(gw))
+            gateways_retrieved = True
+        else:
+            gateways_retrieved = True
+            if nat_gateway_id:
+                if DRY_RUN_GATEWAYS[0]['nat_gateway_id'] == nat_gateway_id:
+                    existing_gateways = DRY_RUN_GATEWAYS
+            elif subnet_id:
+                if DRY_RUN_GATEWAYS[0]['subnet_id'] == subnet_id:
+                    existing_gateways = DRY_RUN_GATEWAYS
+            err_msg = '{0} Retrieving gateways'.format(DRY_RUN_MSGS)
+
+    except botocore.exceptions.ClientError as e:
+            err_msg = str(e)
+
+    return gateways_retrieved, err_msg, existing_gateways
+
+def wait_for_status(client, wait_timeout, nat_gateway_id, status,
+                    check_mode=False):
+    """Wait for the Nat Gateway to reach a status
+    Args:
+        client (botocore.client.EC2): Boto3 client
+        wait_timeout (int): Number of seconds to wait, until this timeout is reached.
+        nat_gateway_id (str): The Amazon nat id.
+        status (str): The status to wait for.
+            examples. status=available, status=deleted
+
+    Basic Usage:
+        >>> client = boto3.client('ec2')
+        >>> subnet_id = 'subnet-12345678'
+        >>> allocation_id = 'eipalloc-12345678'
+        >>> wait_for_status(client, subnet_id, allocation_id)
+        [
+            true,
+            "",
+            {
+                "nat_gateway_id": "nat-123456789",
+                "subnet_id": "subnet-1234567",
+                "nat_gateway_addresses": [
+                    {
+                        "public_ip": "55.55.55.55",
+                        "network_interface_id": "eni-1234567",
+                        "private_ip": "10.0.0.102",
+                        "allocation_id": "eipalloc-12345678"
+                    }
+                ],
+                "state": "deleted",
+                "create_time": "2016-03-05T00:33:21.209000+00:00",
+                "delete_time": "2016-03-05T00:36:37.329000+00:00",
+                "vpc_id": "vpc-12345677"
+            }
+        ]
+
+    Returns:
+        Tuple (bool, str, list)
+    """
+    polling_increment_secs = 5
+    wait_timeout = time.time() + wait_timeout
+    status_achieved = False
+    nat_gateway = list()
+    states = ['pending', 'failed', 'available', 'deleting', 'deleted']
+    err_msg = ""
+
+    while wait_timeout > time.time():
+        try:
+            gws_retrieved, err_msg, nat_gateway = (
+                get_nat_gateways(
+                    client, nat_gateway_id=nat_gateway_id,
+                    states=states, check_mode=check_mode
+                )
+            )
+            if gws_retrieved and nat_gateway:
+                nat_gateway = nat_gateway[0]
```

Comment:
This is a little odd to me. Esp since the first nat_gateway is a list and the second is a dict.
Maybe 'nat_gateways' and 'nat_gateway' ?